### PR TITLE
ansible module for CoreOS fleet v0.8.3 support

### PR DESCRIPTION
**Pull request by coderfi**

### CODE REVIEW

**Comment by jwerak** [comment]
Hi, thanks for updating to latest fleet version :) 
Could you please add unit to by_keys tuple? I am using it to find on which host is particular unit running.

File: cloud/fleet.py [hunk 1]
```diff
@@ -0,0 +1,418 @@
+#!/usr/bin/python
+
+DOCUMENTATION = '''
+---
+module: fleet
+version_added: "1.8"
```
Comment (resmo) [inline cloud/fleet.py]: update version_added to 1.9

**Comment by johnraz** [comment]
:+1: would be super useful to me :-)

**Comment by moperacz** [comment]
+1

**Comment by eik3** [comment]
@mpdehaan could you give feedback on this PR, what has to be done to make this mergeable?

**Comment by bcoca** [comment]
patience, we are working through the backlog

**Comment by ahjohannessen** [comment]
:+1:

**Comment by jalev** [comment]
+1

**Comment by bcoca** [comment]
so after quick review, I believe this should at least be 2 modules if not more, a fleet_facts to gather information and another to set states.

A couple of other things stood out to me, the use of 'command' is imperative and not declarative (which we strive for in ansible) where we prefer to use 'state'. The other seems to be for handling your own ssh connection issues and tunneling that seem to belong in the ssh connection plugin.

**Comment by JeanMertz** [comment]
I'm looking forward to Fleet support in Ansible.

However, I saw this comment in the code:

```
# this is somewhat hacky, too bad fleetctl commands do not
# provide machine parseable output (yet)
```

I didn't really scan the code any further, but reading that comment, I'd say it makes sense to actually use the [fleet API](https://github.com/coreos/fleet/blob/master/Documentation/api-v1.md) or a [Python client](https://github.com/cnelson/python-fleet), instead of calling the CLI and parsing its (non-machine parsable) output.

**Comment by coderfi** [comment]
I considered that, however the ansible development guide explicitly (or at
least used to) heavily discourage third party dependencies. :(

**Comment by bcoca** [comment]
3rd party dependencies are discouraged for ansible itself, the modules have a lot more leeway, specifically because it is needed. i.e. (boto for aws). If there the API or python client make this a better/more stable module we are all for it, it just needs to be specified in the module docs and detect when it is missing and give a clear message that it is a requirement.

**Comment by coderfi** [comment]
Ah, I developed this module before the ansible extras repo was split out
from the main ansible repo (that's how long it has been waiting in the
queue for acceptance!)

It makes sense that there is more leeway now.

I'll refactor the module to use the python library, since programmatic
access to fleet is better than scraping some shell command output! :)

Fi

**Comment by gregdek** [comment]
_scratching old comment_

If you can fix the issues that @bcoca brought up, we can merge.

**Comment by gregdek** [comment]
Hey @coderfi -- still interested in refactoring this module?

**Comment by coderfi** [comment]
Hello,

Oops, sorry, this pull request kind of dropped off my radar. :)

That's great that I can use additional third party libraries rather than resorting to string comparison hacks.

I'll try to get to this next weekend. Looks like I should split this into two or more modules , I'll also look at updating to supporting the latest fleet version while I'm at it.

thanks,
Fi

**Comment by gregdek** [comment]
Thanks @coderfi!

**Comment by gregdek** [comment]
If you do decide to split into multiple PRs, please close this one and refer to it in subsequent PRs.

**Comment by coderfi** [comment]
Will do!

**Comment by coderfi** [comment]
Closing this PR, obsoleted by https://github.com/ansible/ansible-modules-extras/pull/929 and https://github.com/ansible/ansible-modules-extras/pull/928

Note: these PRs are based on the python fleet library. Unfortunately, it lacks a few of the capabilities present in fleetctl (and this PR), such as 'journal'.

**Pull request closed without merging**